assets/src/js/woocommerce/admin/wc-product-video-gallery.js

```
@@ -0,0 +1,149 @@
+								`<input type="hidden"
+									name="rtgodam_product_video_gallery_urls[]"
+									value="${ data.url }" />`,
+							);
+
+							const $thumbWrapper = $( '<div>', {
+								class: 'video-thumb-wrapper',
+							} );
+
+							if ( thumbnail ) {
+								$thumbWrapper.append(
+									`<img src="${ thumbnail }" alt="${ __( 'Video thumbnail', 'godam' ) }" class="godam-video-thumbnail" />`,
+								);
+							} else {
+								$thumbWrapper.append(
+									`<span style="color:#aaa; margin-bottom: 10px; display:block; aria-label="${ __( 'No thumbnail available', 'godam' ) }"">${ __( 'No thumbnail available', 'godam' ) }</span>`,
+								);
+							}
+
+							const $videoTitle = $( '<div>', {
+								class: 'godam-product-video-title',
+								text: videoTitle,
+								title: videoTitle,
+							} );
+
+							$thumbWrapper.append(
+								$( '<button>', {
+									type: 'button',
+									class: 'godam-remove-video-button components-button godam-button is-compact is-secondary has-icon wc-godam-product-admin',
+									'aria-label': __( 'Remove video from gallery', 'godam' ),
+									html: `
+									<img src="${ RTGodamVideoGallery.DeleteIcon }" alt="${ __( 'Delete Bin Icon', 'godam' ) }" width="14" height="14" style="vertical-align:middle;" />
+								`,
+								} ),
+							);
+
+							listItem.append( $thumbWrapper ).append( $videoTitle ).append( $addBtn );
+							videoList.append( listItem );
+
+							wp.apiFetch( {
+								path: `${ RTGodamVideoGallery.namespace }${ RTGodamVideoGallery.videoCountEP }/${ data.id }`,
+							} )
+								.then( ( res ) => {
+									updateAddProductButtonLabel( $addBtn, res.count );
+
+									if ( Array.isArray( res.linked ) ) {
+										$addBtn.attr(
+											'data-linked-products',
+											JSON.stringify( res.linked ),
+										);
+									}
+								} )
+								.catch( () => {
+									// If endpoint fails just leave the default label & an empty list.
```

If something happens which is not expected, we can't silently fail. We need to show error message on why it failed.